src/pages/LogView/LoadingPage/useResolveLogURL.ts

```diff
@@ -0,0 +1,121 @@
+import { useQuery } from "@apollo/client";
+import { LogTypes } from "constants/enums";
+import {
+  getJobLogsURL,
+  getLegacyJobLogsURL,
+} from "constants/externalURLTemplates";
+import {
+  getEvergreenTaskLogURL,
+  getEvergreenTestLogURL,
+  getLobsterResmokeURL,
+  getLobsterTaskURL,
+  getLobsterTestURL,
+  getResmokeLogURL,
+} from "constants/logURLTemplates";
+import { TestLogUrlQuery, TestLogUrlQueryVariables } from "gql/generated/types";
+import { GET_TEST_LOG_URL } from "gql/queries";
+
+interface Props {
+  buildID?: string;
+  execution?: string;
+  groupID?: string;
+  logType: string;
+  origin?: string;
+  taskID?: string;
+  testID?: string;
+}
+
+export const useResolveLogURL = ({
+  buildID,
+  execution,
+  groupID,
+  logType,
+  origin,
+  taskID,
+  testID,
+}: Props) => {
+  const { data: testData, loading: isLoadingTest } = useQuery<
+    TestLogUrlQuery,
+    TestLogUrlQueryVariables
+  >(GET_TEST_LOG_URL, {
+    variables: {
+      testName: `^${testID}$`,
+      taskID: taskID as string,
+      execution: parseInt(execution as string, 10),
+    },
+    skip: !(
```

Comment:
One of the biggest motivations of https://jira.mongodb.org/browse/EVG-19354 is that all test logs would originate from the same source and we would resolve the logs the same way from the test results resolver. So this should apply to both resmoke and normal test logs.